packages/eslint-plugin-mdx/src/processors/options.ts

```diff
@@ -49,6 +49,9 @@ ESLinter.prototype.verify = function (
     {}
 
   processorOptions.lintCodeBlocks = settings['mdx/code-blocks'] === true
+  processorOptions.languageMapper = settings[
+    'mdx/language-mapper'
+  ] as ProcessorOptions['languageMapper']
```

Comment:
Would it make sense for `languageMapper` to have it's own type/interface which could be used directly in this cast?

Reply:
Added a `ESLintMdxSettings` interface

https://github.com/mdx-js/eslint-mdx/pull/291/files#diff-942bc9ad585a6cef970e63062046e66eebb59a85e131d0894e5b7db3391541baR20-R23

But after digging some other plugins like https://github.com/yannickcr/eslint-plugin-react#configuration or https://github.com/mysticatea/eslint-plugin-node/blob/master/docs/rules/no-extraneous-import.md#shared-settings, a namespaced object settings could be used easyier.

```json
{
  "settings": {
    "mdx": {
      "codeBlocks": true,
      "languageMapper": {}
    }
  }
}
```

And I don't know if it should be considered as BREAKING CHANGE, because it's very easy to migrate.

Reply:
Or wait for https://github.com/eslint/rfcs/tree/master/designs/2019-processor-shared-settings first.